pkg/config/nacos/nacos.go

```diff
@@ -0,0 +1,275 @@
+		vo.NacosClientParam{
+			ServerConfigs: serverConfigs,
+			ClientConfig:  &clientConfig,
+		},
+	)
+
+	if err != nil {
+		return err
+	}
+	s.client = client
+	return nil
+}
+
+func parseServerConfig(options map[string]interface{}) []constant.ServerConfig {
+	cfgs := make([]constant.ServerConfig, 0)
+
+	scheme := "http"
+	if val, ok := options[_scheme]; ok {
+		scheme = val.(string)
+	}
+	contextPath := "/nacos"
+	if val, ok := options[_contextPath]; ok {
+		contextPath = val.(string)
+	}
+
+	if servers, ok := options[_server]; ok {
+		addresses := strings.Split(servers.(string), ",")
+		for i := range addresses {
+			addr := strings.Split(strings.TrimSpace(addresses[i]), ":")
+
+			ip := addr[0]
+			port, _ := strconv.ParseInt(addr[1], 10, 64)
+
+			cfgs = append(cfgs, constant.ServerConfig{
+				Scheme:      scheme,
+				ContextPath: contextPath,
+				IpAddr:      ip,
+				Port:        uint64(port),
+			})
+		}
+	}
+
+	return cfgs
+}
+
+func parseClientConfig(options map[string]interface{}) constant.ClientConfig {
+	cc := constant.ClientConfig{}
+
+	if val, ok := options[_namespaceKey]; ok {
+		cc.NamespaceId = val.(string)
+	}
+	if val, ok := options[_username]; ok {
+		cc.Username = val.(string)
+	}
+	if val, ok := options[_password]; ok {
+		cc.Password = val.(string)
+	}
+	return cc
+}
+
+func (s *storeOperate) loadDataFromServer() error {
+	defer s.cfgLock.Unlock()
+	s.cfgLock.Lock()
+
+	for dataId := range config.ConfigKeyMapping {
+		data, err := s.client.GetConfig(vo.ConfigParam{
+			DataId: string(dataId),
+			Group:  s.groupName,
+		})
+		if err != nil {
+			return err
+		}
+
+		s.confMap[dataId] = data
+	}
+
+	return nil
+}
+
+//Save save a configuration data
+func (s *storeOperate) Save(key config.PathKey, val []byte) error {
+
+	_, err := s.client.PublishConfig(vo.ConfigParam{
+		Group:   s.groupName,
+		DataId:  string(key),
+		Content: string(val),
+	})
+
+	return err
+}
+
+//Get get a configuration
+func (s *storeOperate) Get(key config.PathKey) ([]byte, error) {
+	defer s.cfgLock.RUnlock()
+	s.cfgLock.RLock()
+
+	val := []byte(s.confMap[key])
+	return val, nil
+}
+
+//Watch Monitor changes of the key
+func (s *storeOperate) Watch(key config.PathKey) (<-chan []byte, error) {
+	defer s.lock.Unlock()
+	s.lock.Lock()
+	if _, ok := s.receivers[key]; !ok {
+		w, err := s.newWatcher(key, s.client)
+		if err != nil {
+			return nil, err
+		}
+
+		ctx, cancel := context.WithCancel(context.Background())
+		go w.run(ctx)
+		s.cancelList = append(s.cancelList, cancel)
+		s.receivers[key] = w
+	}
+
+	w := s.receivers[key]
+
+	defer w.lock.Unlock()
+	w.lock.Lock()
+
+	rec := make(chan []byte)
+	s.receivers[key].receivers = append(s.receivers[key].receivers, rec)
+	return rec, nil
+}
+
+//Name plugin name
+func (s *storeOperate) Name() string {
+	return "nacos"
+}
+
+//Close do close storeOperate
+func (s *storeOperate) Close() error {
+	return nil
+}
+
+type nacosWatcher struct {
+	lock      *sync.RWMutex
```

Comment:
Use `sync.RWMutex` directly, no need to use pointer type, and no need to initialize manually.